frontend/public/components/utils/container-select.tsx

```diff
@@ -0,0 +1,92 @@
+import * as React from 'react';
+import * as _ from 'lodash-es';
+import {
+  Divider,
+  MenuToggle,
+  MenuToggleElement,
+  Select,
+  SelectGroup,
+  SelectList,
+  SelectOption,
+} from '@patternfly/react-core';
+import { useTranslation } from 'react-i18next';
+
+import { ContainerModel } from '@console/internal/models';
+import { ContainerSpec } from '@console/internal/module/k8s';
+import { ResourceName } from './resource-icon';
+
+export const getContainerLabel = (name: string) => (
+  <ResourceName name={name} kind={ContainerModel.kind} />
+);
+
+const getSelectOptions = (containers: Containers) =>
+  Object.values(containers).map(({ name }) => (
+    <SelectOption key={name} value={name}>
+      {getContainerLabel(name)}
+    </SelectOption>
+  ));
+
+export const ContainerSelect: React.FC<ContainerSelectProps> = ({
+  containers,
+  currentKey = '',
+  initContainers = {},
```

Comment:
nit, avoid using destructured default values since they are not null safe. Instead, use optional chaining, nullish coalescing operator, and boolean or ternary expressions when referencing optional props.